src/main/java/net/ajpappas/discord/modules/audiostreamer/commands/QueueCommand.java

```diff
@@ -31,6 +31,42 @@ public static void queueCommand(CommandContext cc) {
         List<AudioTrack> queue = AudioStreamer.getVoiceManager().getQueue(cc.getGuild());
         AudioTrack nowPlaying = AudioStreamer.getVoiceManager().getNowPlaying(cc.getGuild());
 
+        //!queue remove <index>
+        if (cc.getArgCount() > 1 && cc.getArgument(1).equalsIgnoreCase("remove")) {
+            //user must have
+            if (!AudioStreamer.hasDJPermissions(cc.getUser(), cc.getGuild())) {
+                cc.replyWith("You do not have permissions to use this command.");
+                return;
+            }
+            final String usage = "usage: !queue remove [1-X]";
+            if (cc.getArgCount() != 3) {
+                cc.replyWith(usage);
+                return;
+            }
+            int songIndex;
+            try {
+                songIndex = Integer.parseInt(cc.getArgument(2));
+            } catch (NumberFormatException e) {
+                cc.replyWith(cc.getArgument(2) + " is not a valid song index.");
+                return;
+            }
+            if (songIndex < 1 || songIndex > queue.size() + 1) {
+                cc.replyWith(cc.getArgument(2) + " is not a valid song index.");
```

Comment:
Make it more user friendly and say `"\" + cc.getArgument(2) + "\" is not a valid song number."`

Reply:
It has to be queue.size()+1 because it has to accept all songs in the queue plus the currently playing song